admin/server/projects.go

```diff
@@ -124,14 +136,20 @@ func (s *Server) GetProject(ctx context.Context, req *adminv1.GetProjectRequest)
 	}
 
 	claims := auth.GetClaims(ctx)
-	if !claims.Can(ctx, proj.OrganizationID, auth.ReadProjects, proj.ID, auth.ReadProject) {
+	permissions := claims.ProjectPermissions(ctx, proj.OrganizationID, proj.ID)
+	if proj.Public {
+		permissions.ReadProject = true
+		permissions.ReadProd = true
+	}
+
+	if !permissions.ReadProject {
 		return nil, status.Error(codes.PermissionDenied, "does not have permission to read project")
 	}
 
-	if proj.ProdDeploymentID == nil {
+	if proj.ProdDeploymentID == nil || !permissions.ReadProd {
```

Comment:
should this be `permissions.ReadProd` ? whats the meaning of `!permissions.ReadProd` ?

Reply:
If this `if` clause is triggered, it will only return the project info, not the prod deployment info and JWT. So if user only has `ReadProject`, but not `ReadProd` (kind of weird, but possible with current permissions), they will get the project info, but not the prod deployment info.